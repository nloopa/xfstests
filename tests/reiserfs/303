#! /bin/bash
# FS QA Test No. 303
#
# Test to verify xfs_quota(8) administrator commands can deal with invalid
# storage mount point without NULL pointer dereference problem.
#
#-----------------------------------------------------------------------
# Copyright (c) 2013 Oracle Inc.  All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#-----------------------------------------------------------------------
#

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1	# failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 15

_cleanup()
{
    cd /
    rm -f $tmp.*
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter

# real QA test starts here
_supported_fs xfs
_supported_os IRIX Linux

echo "Silence is golden"

# To run this sanity check we don't need to enable quota function
# nor need a scratch device as quota tool will get the mount path
# at first.
# An non-existent xfs mount point
INVALID_PATH="/INVALID_XFS_MOUNT_POINT"

xfs_quota -x -c 'report -a' $INVALID_PATH	2>/dev/null
xfs_quota -x -c 'state -a' $INVALID_PATH	2>/dev/null
xfs_quota -x -c 'free -h' $INVALID_PATH		2>/dev/null
xfs_quota -x -c 'quot -v' $INVALID_PATH		2>/dev/null
xfs_quota -x -c 'remove' $INALID_PATH		2>/dev/null
xfs_quota -x -c 'disable' $INVALID_PATH		2>/dev/null
xfs_quota -x -c 'enable' $INVALID_PATH		2>/dev/null

# success, all done
status=0
exit
