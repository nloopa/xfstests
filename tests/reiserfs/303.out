QA output created by 303
Silence is golden
